### ISSUE TITLE
Ruleset setting takes precedence over .editorconfig

### ISSUE THREAD

**Issue by julianbartel** [dotnet/roslyn#55920]
According to https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2019:
> Setting a rule's severity in an EditorConfig file takes precedence over any severity that's set in a rule set or in Solution Explorer.

It looks like this is not true, so I decided to report that as a bug.

**Version Used**: 
- VS 16.11.1
- `csc -version` 3.11.0-4.21403.6 (ae1fff34)
- `msbuild -version` 16.11.0.36601

**Steps to Reproduce**:

1. Create a new project, targeting `net4.8` (tried with both SDK style project and classical project)
2. Add following ruleset and add property `<CodeAnalysisRuleSet>ProjectRules.ruleset</CodeAnalysisRuleSet>` to project file:
```xml
<?xml version="1.0" encoding="utf-8"?>
<RuleSet Name="Project rule set" ToolsVersion="16.0">
  <Rules AnalyzerId="Microsoft.CodeAnalysis.NetAnalyzers" RuleNamespace="Microsoft.CodeAnalysis.NetAnalyzers">
    <Rule Id="CA1305" Action="Warning" />
  </Rules>
</RuleSet>
```

2. Add this .editorconfig to the project
```editorconfig
[*.cs]
dotnet_diagnostic.CA1305.severity = suggestion
```
3. Add a code file with the following expression: `1.ToString()`

**Expected Behavior**:
CA1305 is reported as a suggestion in IDE and not reported in command line build with `msbuild`.

**Actual Behavior**:
CA1305 is reported as a warning.

When I remove the line in the ruleset, it changes from a warning to suggestion what tells me that the .editorconfig file is considered.

**Comment by sharwell** [comment]
@julianbartel I'm fairly certain the documentation is not correct. Setting the rule's severity in a [Global AnalyzerConfig](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#global-analyzerconfig) will take precedence over a ruleset file, but both of these take precedence over a normal EditorConfig file.

/cc @mavasani @mikadumont

**Comment by julianbartel** [comment]
@sharwell @mavasani Thank you both for your clarifying answers. I'll investigate the transition to AnalyzerConfig. Sounds like the way to go.

One more question:
Your answer _"[...] both of these take precedence over a normal EditorConfig file"_ confuses me a bit: The table in https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#precedence says

> <html>
> <body>
> 
> 
> In an EditorConfig file and a Global AnalyzerConfig file | The entry in the EditorConfig file wins.
> -- | --
> 
> 
> 
> </body>
> </html>

Also the [design issue](https://github.com/dotnet/roslyn/issues/47707) says, that the EditorConfig wins. 

If documentation is correct here, I am fine. Otherwise, I wonder how to ship a default configuration with a NuGet package and allow developers to overwrite those defaults in their projects? Did I misunderstand something more?

**Comment by mavasani** [comment]
> If documentation is correct here, I am fine

Yes, the documentation is indeed correct. It was written after extensively discussing the severity precedence rules with the compiler team. It also has extensive unit tests, so any regression would be caught. Closing the issue.

**Comment by julianbartel** [comment]
@mavasani Sorry to come back to this again:

Though the documentation referred in my last post may be correct, but the documentation referred in my initial post 
> According to https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2019:
> >Setting a rule's severity in an EditorConfig file takes precedence over any severity that's set in a rule set or in Solution Explorer.

seems not.
